packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,782 @@
+import { Base64 } from 'js-base64';
+import { first, flow, partial, result, trim } from 'lodash';
+import {
+  localForage,
+  APIError,
+  ApiRequest,
+  unsentRequest,
+  responseParser,
+  Entry,
+  AssetProxy,
+  PersistOptions,
+  readFile,
+  CMS_BRANCH_PREFIX,
+  generateContentKey,
+  parseContentKey,
+  labelToStatus,
+  isCMSLabel,
+  EditorialWorkflowError,
+  statusToLabel,
+} from 'netlify-cms-lib-util';
+
+export const API_NAME = 'Azure DevOps';
+
+export class AzureRepo {
+  org: string;
+  project: string;
+  name: string;
+
+  constructor(location?: string | null) {
+    if (!location || location == undefined) {
+      throw new Error(
+        "An Azure repository must be specified in the format 'organisation/project/repo'.",
+      );
+    }
+
+    const components = trim(location, '/').split('/', 3);
+    this.org = components[0];
+    this.project = components[1];
+    this.name = components[2] || components[1];
+  }
+}
+
+export interface AzureUser {
+  id: string;
+  displayName: string;
+  emailAddress: string;
+}
+
+export interface AzureCommitAuthor {
+  name: string;
+  email: string;
+}
+
+enum AzureCommitChangeType {
+  ADD = 'add',
+  DELETE = 'delete',
+  RENAME = 'rename',
+  EDIT = 'edit',
+}
+
+enum AzureCommitContentType {
+  RAW = 'rawtext',
+  BASE64 = 'base64encoded',
+}
+
+enum AzurePullRequestStatus {
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+  ABANDONED = 'abandoned',
+}
+
+type AzureRefUpdate = {
+  name: string;
+  oldObjectId: string;
+};
+
+class AzureRef {
```

Comment:
Can we use plain JavaScript objects instead of classes?
Looks like these classes are just containers for properties.